sdk/src/types/block/output/feature/metadata.rs

```diff
@@ -27,77 +64,105 @@ impl MetadataFeature {
 
     /// Creates a new [`MetadataFeature`].
     #[inline(always)]
-    pub fn new(data: impl Into<Vec<u8>>) -> Result<Self, Error> {
-        Self::try_from(data.into())
+    pub fn new(data: impl IntoIterator<Item = (Vec<u8>, Vec<u8>)>) -> Result<Self, Error> {
```

Comment:
Maybe we should just wrap the inner map methods instead and have `new` create an empty map

Reply:
> Maybe we should just wrap the inner map methods instead

What do you mean by that?

Reply:
I meant provide an API like this:

```rust
impl MetadataFeature {
  pub fn new() -> Self {
    Self(Default::default())
  }

  pub fn with_value(mut self, key: &str, value: impl Into<Vec<u8>>) -> Result<Self, Error> {
    if self
        .deref_mut()
        .insert(
            key.as_bytes().to_vec().into_boxed_slice()
                .try_into()
                .map_err(Error::InvalidMetadataFeatureKeyLength)?,
            value.into().into_boxed_slice()
                .try_into()
                .map_err(Error::InvalidMetadataFeatureValueLength)?,
        )
        .is_some()
    {
        return Err(Error::InvalidMetadataFeature("Duplicated metadata key".to_string()));
    }
    Ok(self)
  }
}

...

MetadataFeature::new().with_value("data1", metadata1)?.with_value("data2", metadata2)?
```

Reply:
I don't really mind, I guess we'll get more experience with it as we use it and we'll see what makes sense in terms of API. I'm not sure why we'd need to construct an empty map if we already have the values, but a function to add values (if we don't have one) is a good idea.

Reply:
Just tried and deref_mut() is not implemented for the type, so we don't have a way to do this?

Reply:
that seems like a big problem for the usability of the type, probably need to update packable

Reply:
alternative: use `BTreeSet<String, Vec<u8>>` in the feature and only use the packable types when packing/unapcking. We have to do it manually anyway so it's not such a big deal imo

Reply:
I think it we removed it since the bounds aren't checked if you just insert some value
For a safe way we would probably need custom wrapper methods that do the validation?

Reply:
Tbh I think we should just merge as is and revisit later 🤷🏻‍♂️

Reply:
alright 🤷‍♀️